android/src/androidTest/java/bbct/android/common/navigation/test/NavigateUpFromAboutTest.java

```diff
@@ -0,0 +1,127 @@
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+
+
+@RunWith(AndroidJUnit4.class)
+abstract public class NavigateUpFromAboutTest<T extends MainActivity> {
+    @Rule
+    public ActivityTestRule<T> activityActivityTestRule;
+    private Context context = null;
+    private DatabaseUtil dbUtil = null;
+
+    public NavigateUpFromAboutTest(Class<T> activityClass) {
+        this.activityActivityTestRule = new ActivityTestRule<>(activityClass);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        activityActivityTestRule.getActivity()
+                .getSupportFragmentManager().beginTransaction();
+        context = getInstrumentation().getTargetContext();
+        this.dbUtil = new DatabaseUtil(context);
+    }
+
+    @Test
+    public void testNavigateToBaseballCardDetailsThenAboutFragment() {
+        String expectedTitle = context.getString(R.string.app_name);
+
+        // pop to BaseballCardList first if db is empty
+        if (this.dbUtil.isEmpty()) {
+            onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        }
+        // navigate to BaseballCardDetails
+        onView(allOf(withContentDescription(R.string.add_menu), isDisplayed())).perform(click());
+
+        // navigate to About
+        navigateToAboutViaContextMenu();
+
+        // Navigate Up
+        onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        onView(allOf(withText(expectedTitle), isDisplayed()));
+    }
+
+    @Test
+    public void testNavigateToFilterThenAboutFragment() {
+        String expectedTitle = context.getString(R.string.app_name);
+
+        // pop to BaseballCardList first if db is empty
+        if (this.dbUtil.isEmpty()) {
+            onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        }
+
+        // navigate to FilterCards
+        onView(allOf(withContentDescription(R.string.filter_cards_title), isDisplayed())).perform(click());
+
+        navigateToAboutViaContextMenu();
+
+        // Navigate Up
+        onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        onView(allOf(withText(expectedTitle), isDisplayed()));
+    }
+
+    @Test
+    public void testNavigateToAboutFragment() {
+        navigateToAboutViaContextMenu();
+        String aboutTitle = context.getString(R.string.about_title);
+        String expectedTitle = context.getString(R.string.app_name, aboutTitle);
+        onView(withText(expectedTitle)).check(matches(isDisplayed()));
```

Comment:
Even shorter lines like this should be broken on each chained call.

Reply:
I see these three lines repeated in `testNavigateUp()` but not in `testNavigateToFilterThenAboutFragment()`. Is there a reason for that or is it an oversight? Does it make sens to move these three lines to `navigateToAboutViaContextMenu()` so they don't have to be repeated?